python/bind_operators.h

```diff
@@ -20,6 +20,28 @@ void bind_comparison(pybind11::class_<T, Ignored...> &c) {
       py::call_guard<py::gil_scoped_release>());
 }
 
+template <class Other, class T, class... Ignored>
+void bind_variable_comparison(pybind11::class_<T, Ignored...> &c) {
```

Comment:
Why not named `bind_comparison`, and why using explicit `variable::equal` (etc.) below? Can't we use `equal` and rely on ADL, such that this function will also work once we define `equal` for, e.g., `DataArray`?